### PR TITLE
Platform selection for D2 ratings.

### PR DESCRIPTION
**Pull request by 48klocs**
In the settings menu, if you're working with a Destiny 2 character and either ratings or reviews are turned on, we also present you the option of filtering down what platform(s) you want to get reviews from. The API was modified to support the options you see.

If no platform is selected (which is the default), it returns reviews from people on all platforms, same as it does now.

shouts-to-my-mouse-heads

### CODE REVIEW

File: src/app/settings/settings.html [hunk 1]
```diff
@@ -84,6 +84,17 @@ <h1 ng-i18next="Settings.Settings"></h1>
...
       </tr>
+      <tr ng-if="vm.settings.destinyVersion === 2 && (vm.settings.showReviews || vm.settings.allowIdPostToDtr)">
+        <td>
+          <label for="reviewPlatformSelection" ng-i18next="Settings.PlatformSelection"></label>
+        </td>
+        <td>
+          <select id="reviewPlatformSelection"
+            ng-model="vm.settings.platformSelection"
+            ng-options="code as name for (code, name) in vm.platformOptions"
+            ng-change="vm.platformChanged()"></select>
```
Comment (bhollis) [inline src/app/settings/settings.html]: Same with the other "platforms".

File: src/app/settings/settings.html [hunk 2]
```diff
@@ -84,6 +84,17 @@ <h1 ng-i18next="Settings.Settings"></h1>
           <input type="checkbox" id="allowIdPostToDtr" ng-model="vm.settings.allowIdPostToDtr"/>
         </td>
       </tr>
+      <tr ng-if="vm.settings.destinyVersion === 2 && (vm.settings.showReviews || vm.settings.allowIdPostToDtr)">
+        <td>
+          <label for="reviewPlatformSelection" ng-i18next="Settings.PlatformSelection"></label>
+        </td>
+        <td>
+          <select id="reviewPlatformSelection"
+            ng-model="vm.settings.platformSelection"
```
Comment (bhollis) [inline src/app/settings/settings.html]: Would you mind changing this to `reviewsPlatformSelection`?

File: src/app/settings/settings.html (hunk 1 above)
Comment (48klocs) [reply src/app/settings/settings.html]: The plural above I follow, but I don’t understand this.

Comment (bhollis) [reply src/app/settings/settings.html]: `vm.settings.reviewsPlatformSelection`, `vm.reviewsPlatformOptions`, `vm.reviewsPlatformChanged()`.

File: src/app/destinyTrackerApi/d2-reviewsFetcher.js [hunk 3]
```diff
@@ -20,23 +20,25 @@ class D2ReviewsFetcher {
...
   }
 
-  _getItemReviewsCall(item) {
+  _getItemReviewsCall(item, platformSelection) {
+    const queryString = `reviews?page=1&platform=${platformSelection}`;
+
     return {
       method: 'POST',
-      url: 'https://db-api.destinytracker.com/api/external/reviews?page=1', // TODO: pagination
+      url: `https://db-api.destinytracker.com/api/external/reviews?${queryString}`, // TODO: pagination
```
Comment (dbuck) [inline src/app/destinyTrackerApi/d2-reviewsFetcher.js]: `reviews?` in both the url and the queryString

Comment (48klocs) [reply src/app/destinyTrackerApi/d2-reviewsFetcher.js]: Thanks for noticing!

**Pull request merged**